qcodes/dataset/data_set_cache.py

```diff
@@ -66,6 +74,16 @@ def load_data_from_db(self) -> None:
             self._read_status,
             self._data
         )
+        if old_write_status != self._write_status:
+            self._live = False
+
+    @property
+    def live(self) -> Optional[bool]:
```

Comment:
it's very interesting that with this implementation the cache can be loaded from the database and THEN extra data can be added to it (and once that is done, reading from database is not possible anymore regardless of whether there is new data there or not). What would we use this for? If we are not going to use it, i wonder if we should restrict the cache to always be either live or not-live, as opposed to give him a change to become live after being not live :)

Reply:
I think you have misunderstood the logic. 

It can be loaded from the db and if that happens live is set to False and you can continue loading from the db until all data has been loaded from the dataset as signaled by the dataset being marked completed. 

Alternatively you can start building the cache from live data in which case it is an error to try to load from the db.

Reply:
to clarify this we should probably raise in live is false when calling `add_data`

Reply:
thanks for the explanation! perhaps it's worth adding it to the docstring of the cache?

> raise in live is false when calling add_data
yes :)

Reply:
both are done